esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -677,21 +779,43 @@ def __init__(
             exceeds ``mdtol``. ``mdtol=0`` means no missing data is tolerated while
             ``mdtol=1`` will mean the resulting element will be masked if and only
             if all the contributing elements of data are masked.
+        method : str, default="conservative"
+            The regridding method which :mod:`ESMF` uses to calculate weights. Can be
+            either "conservative" or "bilinear".
 
         """
+        if method not in ["conservative", "bilinear"]:
+            raise (ValueError("method must be either bilinear or conservative."))
         # Missing data tolerance.
         # Code directly copied from iris.
+        if mdtol is None:
+            if method == "conservative":
+                mdtol = 1
+            elif method == "bilinear":
+                mdtol = 0
         if not (0 <= mdtol <= 1):
             msg = "Value for mdtol must be in range 0 - 1, got {}."
             raise ValueError(msg.format(mdtol))
         self.mdtol = mdtol
+        self.method = method
 
         partial_regrid_info = _regrid_rectilinear_to_unstructured__prepare(
-            src_mesh_cube, target_grid_cube, precomputed_weights=precomputed_weights
+            src_mesh_cube,
+            target_grid_cube,
+            method=method,
```

Comment:
```suggestion
            method=self.method,
```

Less risk of future divergence.